Sources/SwiftProtobuf/Visitor.swift

```diff
@@ -445,8 +448,33 @@ public protocol Visitor {
   mutating func visitUnknown(bytes: Data) throws
 }
 
+/// Provides options for how visitor traversal should be carried out
+public struct TraversalOptions {
+  static let `default` = TraversalOptions()
+
+  /// Determines if non-optional fields that are equal to their default values should be visited.
+  /// Defaults to `false`.
+  ///
+  /// ### When set to true:
+  /// - repeated fields: Will always be visited, even if they are empty.
+  /// - map fields: Will always be visited, even if they are empty.
+  /// - singular message and group fields: Will be visited only if they are set.
+  /// - singular primitive fields: Will always be visited under proto2. Under proto3 unset optional primitive fields will not be visited.
+  public var alwaysVisitPrimitiveFields: Bool
+
+  public init(alwaysVisitPrimitiveFields: Bool = false) {
```

Comment:
@tbkka (and others) - I think we've mostly left off doing an initializer like this because it could be somewhat fragile if options get added.  Should we just skip listing it and do `public var alwaysVisitPrimitiveFields: Bool = false` like we've done in all the other `*Options`?

Reply:
As a user of the library I find it annoying to have to create a `var` in order to be able to set some of the flags. If you want to keep the empty init then would you be opposed to duplicating the `static func with(_ populator: (inout Self) -> ()) rethrows -> Self` pattern from Message here?

Reply:
Added this utility ☝️. LMK if you'd like me to remove it.